### ISSUE TITLE
Add category "APIs" to the options

### ISSUE THREAD

**Issue by TitusKirch** [WoltLab/WCF#3503]
Hello,

I would appreciate it if there was a category "APIs" in the options. Currently, every developer does his own thing and a single place where the end user can enter his API keys/tokens/secrets etc. would make it easier not only for the end user but also for the developers (potentially less support effort).

Post in the forum: https://community.woltlab.com/thread/285221-kategorie-apis-bei-den-optionen-hinzufügen/

**Comment by dtdesign** [comment]
Overall I'm not too happy with the current state of the options. Some option categorizations are arbitrary at best and the more options are being added, the harder it becomes to quickly find the one you're looking for. It gets worse when third party plugins are involved that straight up spawn even more categories to sift through. I can understand the idea behind a unified category for API integrations, but I don't think it is a healthy move.

The decision where to place options remains arbitrary. Fine, credentials go in there, but the frequency to query that API? Does it go into the API section or is it placed in the regular category? Where do you draw the imaginary line that divides where each option goes? How does the user know where to look to find it?

This gets even worse by the galore of options added with even the tiniest plugin, eventually turning a specialized API category into a landfill. History tells me that this will happen ;)

The bottom line is that I have serious doubts about the effectiveness of this category, because it does not solve the underlying problem of having hard to find options. The proposed category could even make it worse by causing previously neatly arranged options to be split into two entirely separated categories. Keeping the options together in this example would turn the API category ad absurdum.

In the past I had tried out a few different concepts for the options, but so far I wasn't satisfied with the results. I'm open for suggestions on this though, maybe someone else has a good idea on how we can enhance the options?

**Comment by TitusKirch** [comment]
I understand the points you have made. Maybe it would make more sense to revise and restructure the options system with the next major version (6.0). Here you could also define exact guidelines for third party providers where what belongs and improve the overview in general.

However, if a restructuring is being considered, I would start a discussion early enough so that all parties can participate in an optimal solution.
